xcube_cci/cciodp.py

```diff
@@ -424,17 +427,39 @@ def get_dataset_info(self, dataset_id: str, dataset_metadata: dict = None) -> di
         if not dataset_metadata:
             dataset_metadata = self.get_dataset_metadata(dataset_id)
         nc_attrs = dataset_metadata.get('attributes', {}).get('NC_GLOBAL', {})
-        data_info['lat_res'] = _get_res(nc_attrs, 'lat')
-        data_info['lon_res'] = _get_res(nc_attrs, 'lon')
+        data_info['crs'] = \
+            self._get_crs(dataset_metadata.get('variable_infos', {}))
+        data_info['y_res'] = _get_res(nc_attrs, 'lat')
+        data_info['x_res'] = _get_res(nc_attrs, 'lon')
         data_info['bbox'] = (float(dataset_metadata['bbox_minx']),
                              float(dataset_metadata['bbox_miny']),
                              float(dataset_metadata['bbox_maxx']),
                              float(dataset_metadata['bbox_maxy']))
-        data_info['temporal_coverage_start'] = dataset_metadata.get('temporal_coverage_start', '')
-        data_info['temporal_coverage_end'] = dataset_metadata.get('temporal_coverage_end', '')
-        data_info['var_names'], data_info['coord_names'] = self.var_and_coord_names(dataset_id)
+        data_info['temporal_coverage_start'] = \
+            dataset_metadata.get('temporal_coverage_start', '')
+        data_info['temporal_coverage_end'] = \
+            dataset_metadata.get('temporal_coverage_end', '')
+        data_info['var_names'], data_info['coord_names'] = \
+            self.var_and_coord_names(dataset_id)
         return data_info
 
+    @staticmethod
+    def _get_crs(variable_infos: dict) -> str:
+        crs = None
+        for var_attrs in variable_infos.values():
+            if 'grid_mapping_name' in var_attrs:
+                crs = pyproj.crs.CRS.from_cf(var_attrs)
```

Comment:
I believe, this is not the right way dealing with the `grid_mapping_name` attribute. The `grid_mapping_name` attribute names another variable whose attributes are used to decode the CRS.

Reply:
I use the grid_mapping_name to identify the crs variable. I found that it is the most reliable source, as the variable might have any name and no other attributes are mandatory. I am open to suggestions for better means of detecting a crs varaible.

Reply:
> The grid_mapping_name attribute names another variable whose attributes are used to decode the CRS.

I was wrong with this. I confused it with the `grid_mapping` attribute.